### ISSUE TITLE
SR04 issue on Tasmota 5.14.0

### ISSUE THREAD

**Issue by alexjoni** [arendst/Sonoff-Tasmota#2763]
I recently added a SR04 distance sensor to a Sonoff basic, and it seems to work ok the first couple of minutes.
After about 5 minutes the readout is 283cm, and it gets stuck to that value.
If I restart the sonoff it works again for 5 minutes then it turns again to 283cm..

Is there a way to debug this further?
Maybe trigger an Sr04Init() function more often?

Regards,
Alex

**Comment by Frogmore42** [comment]
It could also be the SR04 sensor that is bad.  Do you have another one?

**Comment by Frogmore42** [comment]
Since it seems like you are getting the maximum distance, it seems like the sensor is not working well.  Most of them need 5V to work, so that might be the issue.

**Comment by wongnam** [comment]
i had tried before, over 300cm the value will unreadable with Tasmota even your SR04t sensor can read up to 400cm.

**Comment by alexjoni** [comment]
I did use 5V to power it. I'm not sure how clean the 5V on a sonoff basic is, I'll try to add a cap for filtering.
I'll also order a new sensor and see if it makes any difference.

**Comment by helmar74** [comment]
Hi, I've also an issue with 5.14.0 and a HC-SR04.
First of all the sensor is OK. I tried it with an other arduino and it's measuring fine!
I connected the sensor as mentioned here: 
[https://github.com/arendst/Sonoff-Tasmota/wiki/Wemos-D1-Mini-and-HC-SR04-Ultrasonic-Sensor](url)

The problem ist, that the sensor is measuring not correct woth Tasmota FW.
The distance is about 145 cm. 
But within Tasmota the measuring jumps between 96 and 144 cm
Via MQTT sometimes (the correct) 144 cm are sent, somteimes 96 cm.
The SEnsor has free "sight" to the ceiling, so no reflection should occur
So the measuring isn't ok.

Anyone an idea, what to try?
